rpc/src/rpc.rs

```diff
@@ -1898,6 +1898,10 @@ fn verify_transaction(transaction: &Transaction) -> Result<()> {
         return Err(RpcCustomError::TransactionPrecompileVerificationFailure(e).into());
     }
 
+    if transaction.signatures.len() != transaction.message.header.num_required_signatures {
```

Comment:
Any reason not to correct the check in `Transaction`'s `Sanitize` impl?

https://github.com/solana-labs/solana/blob/9bd02c95704faec6e275378fe4a9504618e232be/sdk/src/transaction.rs#L127

Reply:
Yeah, two reasons. First is that sanitization seemed to be more for sanitizing potential out of bounds indices. Second is that the `Sanitize` trait is used here so no easy way to consume the feature switch in there.

Reply:
Sound ok @t-nelson ?

Reply:
Yep! Good points.  `Sanitize` needing plumbed is a pretty hard blocker.

Can we stick the logic in a method on `Transaction`, since we do it twice?

Reply:
Good call, done!